splinterd/src/daemon.rs

```diff
@@ -159,40 +160,78 @@ impl SplinterDaemon {
 
         let store_factory = create_store_factory(&self.db_url)?;
 
+        #[cfg(feature = "deprecate-yaml")]
+        {
+            let circuits_location = Path::new(&self.state_dir).join("circuits.yaml");
+            let proposals_location = Path::new(&self.state_dir).join("circuit_proposals.yaml");
+
+            let circuits_location_exists = circuits_location.exists();
+            let proposals_location_exists = circuits_location.exists();
+
+            if circuits_location_exists || proposals_location_exists {
+                error!("splinterd found yaml state files. run the `splinter upgrade` command to import your yaml state into a database");
+
+                if circuits_location_exists {
+                    error!(
+                        "yaml circuits state file: {}",
+                        circuits_location.to_string_lossy()
+                    );
+                }
+
+                if proposals_location_exists {
+                    error!(
+                        "yaml proposals state file: {}",
+                        proposals_location.to_string_lossy()
+                    );
+                }
+
+                panic!("aborting");
+            }
+        }
+
         let admin_service_store = {
             if let Some(storage) = &self.storage_type {
                 // Get state from the configured storage type and state directory, then
                 // create the new AdminServiceStore
                 match &storage as &str {
                     "yaml" => {
-                        let circuits_location = Path::new(&self.state_dir)
-                            .join("circuits.yaml")
-                            .to_str()
-                            .ok_or_else(|| {
-                                StartError::StorageError(
-                                    "'state_dir' is not a valid UTF-8 string".into(),
-                                )
-                            })?
-                            .to_string();
-                        let proposals_location = Path::new(&self.state_dir)
-                            .join("circuit_proposals.yaml")
-                            .to_str()
-                            .ok_or_else(|| {
-                                StartError::StorageError(
-                                    "'state_dir' is not a valid UTF-8 string".into(),
-                                )
-                            })?
-                            .to_string();
-
-                        Box::new(
-                            YamlAdminServiceStore::new(circuits_location, proposals_location)
-                                .map_err(|err| {
+                        #[cfg(feature = "deprecate-yaml")]
+                        {
+                            error!("splinterd no longer supports yaml state");
+                            panic!("aborting");
```

Comment:
Return StartError here.

Or consider removing this storage option completely.

Reply:
Ah yeah, I think removing the option entirely is probably the cleaner solution